include/mbedtls/ccm.h

```diff
@@ -1,8 +1,10 @@
 /**
  * \file ccm.h
  *
- * \brief CCM combines Counter mode encryption with CBC-MAC authentication
- *        for 128-bit block ciphers.
+ * \brief This file contains CCM definitions and functions.
```

Comment:
Same remark as https://github.com/ARMmbed/mbedtls/pull/1513#discussion_r180687267

Reply:
The information is not removed, but simply moved to the detailed description. The file's brief description should define what the file includes. The assumption that the file name should be sufficient indication is not always correct and can be misleading.

Reply:
Suggestion: "This file provides an API to the CCM block cipher mode which combines counter mode encryption with CBC-MAC authentication for 128-bit block ciphers." @gilles-peskine-arm WDYT?

Reply:
@hanno-arm While this is correct, I think that's too much information for the short description. Most people choose to use the CCM module because they're implementing a protocol that mandates the CCM cipher, and treat CCM as a black box. So the fact that CCM is authenticated encryption is important, but how it's made under the hood is less relevant. How about something like

> This file provides an API for the CCM authenticated encryption mode for block ciphers.

I think this is the right amount information (or maybe “for 128-bit block ciphers”) but I find the phrasing a bit awkward. “for CCM, which is an authenticated encryption mode for block ciphers”?

Reply:
@gilles-peskine-arm Ok, then let's take your suggestion and leave the further explanation to a new line, as it is now - if @RosZad agrees.

Reply:
"This file provides an API for the CCM authenticated encryption mode for block ciphers." is actually a very clear phrasing to indicate that "CCM" is a type of "authenticated encryption mode for block ciphers".